medallion/test/test_mongodb_backend.py

```diff
@@ -1046,17 +1046,17 @@ def test_delete_object(self):
             self.assertEqual(404, r.status_code)
         # ------------- END: test request getting deleted object returns 404 ------------- #
 
-        # ------------- BEGIN: test request getting deleted object manifest returns 404 ------------- #
+        # ------------- BEGIN: test request getting deleted object manifest returns no manifests ------------- #
         r = self.client.get(
             test.MANIFESTS_EP + "?match[id]=%s&match[version]=all" % new_id,
             headers=self.headers,
         )
         objs = self.load_json_response(r.data)
         if r.status_code == 200:
-            self.assertEqual(0, len(objs["objects"]))
+            self.assertEqual(0, len(objs.get("objects", [])))
```

Comment:
Shouldn't we check here for empty envelope as well per the new changes?

Reply:
An empty envelope will pass the test due to the use of `get()` and empty list default.  I guess it doesn't check that the JSON was an object at all, or if it was, that it didn't have other erroneous properties.  But none of the other tests check that either.  What did you have in mind specifically?

Reply:
Hmmm, well none of the other tests in `test_delete_object()` check for empty objects.  I looked again at the changes in this PR (it has been a bit since I wrote them), and I see I did check for empty objects in other test functions (`self.assertEqual(r.json, {})`).

I think in theory, the more correct check would be for a *logically* empty envelope, which means either the empty object, an object with `"objects":[]`, i.e. no objects, one with the previous and `"more": False`, ... There are several ways you can express it, due to optional properties.  I guess I wrote the `{}` checks that way because that's what the mongo backend actually returned, and something more complex didn't seem necessary.  I wrote this check that way probably more because it was a minimal change which did the right thing.

With Zach's experimenting with consolidating backend tests, we might need to be more sophisticated about this, since different backends might produce different but equivalent (and correct) responses, and we need to recognize those variations as being correct.  Maybe we revisit this at that point.  Maybe this test is good enough as it is, for now.

Reply:
Sounds good. Yeah, I foresee a lot of this changing with the consolidation. I was curious because of the other test checking against r.json and {} (empty-dictionary)